src/esp/gfx/PbrShader.h

```diff
@@ -185,28 +190,91 @@ class PbrShader : public Magnum::GL::AbstractShaderProgram {
      * will use the node's semantic ID
      */
     InstancedObjectId = (1 << 12) | ObjectId,
+
+    /**
+     * Has ClearCoat layer.
+     */
+    ClearCoatLayer = 1 << 13,
+    // TODO : possibly break out each layer's flags to a separate flags state
+    // machine?
+    /**
+     * Has ClearCoat Texture in ClearCoat layer
+     */
+    CCLayer_CCTexture = (1 << 14) | ClearCoatLayer,
+    /**
+     * Has Roughness Texture in ClearCoat layer
+     */
+    CCLayer_RoughnessTexture = (1 << 15) | ClearCoatLayer,
+    /**
+     * Has Normal Texture in ClearCoat layer
+     */
+    CCLayer_NormalTexture = (1 << 16) | ClearCoatLayer,
+
+    CCLayer_NormalTextureScale = (1 << 17) | ClearCoatLayer,
+
+    /**
+     * Has KHR_materials_specular layer
+     */
+    SpecularLayer = 1 << 18,
+    // TODO : possibly break out each layer's flags to a separate flags state
+    // machine?
+
+    /**
+     * Has Specular Texture in KHR_materials_specular layer
+     */
+    SpecLayer_SpecTexture = (1 << 19) | SpecularLayer,
+
+    /**
+     * Has Specular Color Texture in KHR_materials_specular layer
+     */
+    SpecLayer_SpecColorTexture = (1 << 20) | SpecularLayer,
```

Comment:
Again, as [the extension states](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_materials_specular/README.md), these two are commonly packed into a single texture, so you could treat them as one.

Unfortunately, as there's no builtin material support in Magnum for `KHR_materials_specular`, there's no easy check if the material indeed has such packing, so you have to do that by hand for now.